modules/core/test/v2/unit/coins/algo.ts

```diff
@@ -64,4 +78,105 @@ describe('ALGO:', function() {
     const pub = algosdk.Address.decode(keyPair.pub).publicKey;
     algosdk.NaclWrapper.verify(message, signature, pub).should.equal(true);
   });
+  
+  describe('Transaction Verification', function() {
+    let basecoin;
+    let wallet;
+    let walletData;
+    let halfSignedTransaction;
+    let prebuild;
+
+    const userKeychain = {
+      pub: 'UMYEHZ2NNBYX43CU37LMINSHR362FT4GFVWL6V5IHPRCJVPZ46H6CBYLYE',
+      prv: 'HEHPVOKEINTQMW4K536BZWUZWQNKBXKIAUSEWCVX3LCXDHI2LVPMV6T5L4'
+    };
+
+    const backupKeychain = {
+      pub: 'LT7DY6JX46K2MUCLJJHEUYWKVC63FXEKCOKUSPGVTDGH3DBBK2V4SEDZWY',
+      // normally this wouldn't be here but we include it for signing purposes
+      prv: 'R4MFSNIAR4PQQGGYA6LK374X6MJEATRVNLREZ3GLAP2VYWLXDN2R4JLI4'
+    };
+
+    const bitgoKeychain = {
+      pub: '3GFMZSPBX3XQL22IBAUF2UQFAZCOF5QFFVRRLEHFPHFLN5NY7WQJSSZRGQ',
+      // normally this wouldn't be here but we include it for signing purposes
+      prv: 'DLHE27PAI335VEQUNX7R2442SSJZUYISBL7OOR3V34AYNTH7YWI6XMGVVE'
+    }
+
+    before(co(function *() {
+      basecoin = bitgo.coin('talgo');
+      walletData = {
```

Comment:
Consider moving some of this to fixtures/

Reply:
k